src/account.ts

```diff
@@ -824,109 +811,139 @@ export default class EthereumAccount {
         this.subBalance(amount)
         this.account.payoutAmount = this.account.payoutAmount.plus(amount)
       } catch (err) {
-          // Balance update likely dropped below the minimum, so throw an internal error
+        // Balance update likely dropped below the minimum, so throw an internal error
         this.master._log.trace(`Failed to fulfill response to PREPARE: ${err.message}`)
         throw new IlpPacket.Errors.InternalError(err.message)
       }
     }
 
-    if (attemptSettle) {
-      this.outgoingSettlements.runExclusive(() => this.attemptSettle())
-        .catch(err => {
-          this.master._log.trace(`Error queueing an outgoing settlement: ${err.message}`)
-        })
+    // Attempt to settle on fulfills and* T04s (to resolve stalemates)
+    const shouldSettle = isFulfill || (isReject && responsePacket.data.code === 'T04')
+    if (shouldSettle) {
+      this.attemptSettle().catch((err: Error) => {
+        this.master._log.trace(`Error queueing an outgoing settlement: ${err.message}`)
+      })
     }
   }
 
-  private startChannelWatcher (): void {
-    const interval = this.master._channelWatcherInterval.toNumber()
-    const timer: NodeJS.Timer = setInterval(async () => {
-      try {
+  private startChannelWatcher () {
+    const timer: NodeJS.Timeout = setInterval(
+      () => this.queue.synchronize(TaskPriority.ChannelWatcher, async () => {
         const claim = this.account.bestIncomingClaim
-        if (!claim) return
+        if (!claim) {
+          // No claim is linked: stop the channel watcher
```

Comment:
Would there be a claim right when it starts up?

Reply:
Yes, if a claim was persisted in the store for that account (e.g. if the server restarted, the channel watcher starts for each account).